src/graph/UpdateEdgeExecutor.cpp

```diff
@@ -0,0 +1,426 @@
+
+    std::vector<cpp2::RowValue> rows;
+    if (rpcResp.__isset.schema && rpcResp.__isset.data) {
+        std::shared_ptr<ResultSchemaProvider> schema;
+        if (rpcResp.get_schema() != nullptr) {
+            schema = std::make_shared<ResultSchemaProvider>(rpcResp.schema);
+        }
+        auto reader = RowReader::getRowReader(rpcResp.data, schema);
+        auto &getters = expCtx_->getters();
+        getters.getAliasProp = [&, this] (const std::string &,
+                                          const std::string &prop) -> OptVariantType {
+            std::string edgeName = *this->edgeTypeName_;
+            auto indexIt = this->schemaPropIndex_.find(std::make_pair(edgeName, prop));
+            if (indexIt == this->schemaPropIndex_.end()) {
+                auto msg = folly::sformat(
+                    "EdgeName : {} , propName : {} is not exist", edgeName, prop);
+                LOG(ERROR) << msg;
+                return Status::Error(msg);
+            }
+            auto index = indexIt->second;
+            const nebula::cpp2::ValueType &type = schema->getFieldType(index);
+            if (type == CommonConstants::kInvalidValueType()) {
+                auto msg =
+                    folly::sformat("Edge: {} no schema for the index {}", edgeName, index);
+                LOG(ERROR) << msg;
+                return Status::Error(msg);
+            }
+            auto res = RowReader::getPropByIndex(reader.get(), index);
+            if (ok(res)) {
+                return value(std::move(res));
+            }
+            return Status::Error(folly::sformat("{}.{} was not exist", edgeName, prop));
+        };
+        getters.getSrcTagProp = [&, this] (const std::string &tagName,
+                                           const std::string &prop) -> OptVariantType {
+            auto tagIt = this->schemaPropIndex_.find(std::make_pair(tagName, prop));
+            if (tagIt == this->schemaPropIndex_.end()) {
+                auto msg = folly::sformat(
+                    "Src tagName : {} , propName : {} is not exist", tagName, prop);
+                LOG(ERROR) << msg;
+                return Status::Error(msg);
+            }
+            auto index = tagIt->second;
+            const nebula::cpp2::ValueType &type = schema->getFieldType(index);
+            if (type == CommonConstants::kInvalidValueType()) {
+                auto msg =
+                    folly::sformat("Tag: {} no schema for the index {}", tagName, index);
+                LOG(ERROR) << msg;
+                return Status::Error(msg);
+            }
+            auto res = RowReader::getPropByIndex(reader.get(), index);
+            if (ok(res)) {
+                return value(std::move(res));
+            }
+            return Status::Error(folly::sformat("{}.{} was not exist", tagName, prop));
+        };
+        std::vector<VariantType> record;
+        record.reserve(yields_.size());
+        for (auto *column : yields_) {
```

Comment:
We don't need to execute the expression again.

Reply:
You are right, execute  all of them at the storage service now!